notmuch.go

```diff
@@ -0,0 +1,273 @@
+package pinbox
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"github.com/jhillyerd/enmime"
+	notmuch "github.com/msp301/go.notmuch"
+)
+
+// Notmuch is a Mailbox object intended to interact with a Notmuch database.
+// See https://notmuchmail.org/ for more information.
+type Notmuch struct {
+	DbPath        string
+	ExcludeLabels []string
+	InboxLabel    string
+	Bundle        []string
+}
+
+// CreateNotmuch creates a new Notmuch object ready to interact with the
+// database specified by the given configuration.
+func CreateNotmuch(config Config) *Notmuch {
+	mailbox := Notmuch{
+		DbPath:        config.Maildir,
+		ExcludeLabels: config.Hidden,
+		InboxLabel:    config.Inbox,
+		Bundle:        config.Bundle,
+	}
+	return &mailbox
+}
+
+func (mailbox *Notmuch) Inbox() ([]interface{}, error) {
```

Comment:
The only reason why `Inbox()` exists is to support bundling emails.
Technically this is just a series of searches that produce results in a slightly different structure than a regular search.
Is there a way for the generic `Mailbox` interface to provide the bundling behaviour itself, so that the underlying implementations don't have to re-implement this each time?

Reply:
Refactored :)